workloads/devstack-xenial.yaml

```diff
@@ -0,0 +1,51 @@
+---
+inherits: xenial
+needs_nested_vm: true
+vm:
+  mem_mib: 6144
+  cpus: 4
+  ports:
+  - host: 55580
+    guest: 80
+...
+---
+{{- define "ENV" -}}
+{{proxyVars .}}
+{{- print " DEBIAN_FRONTEND=noninteractive DEBCONF_NONINTERACTIVE_SEEN=true " -}}
+{{end}}
+{{ define "DEVSTACKPATH" }}{{ if $.MountPath "devstack" }}{{$.MountPath "devstack"}}{{else}}/home/{{.User}}/devstack{{end}}{{end}}
+
+runcmd:
+  - {{beginTask . "Create gitconfig file with rules" }}
+  - su -c 'printf "%s\n"
+    "[url \"http://\"]"
+    "  insteadOf=git://"
+    > /home/{{.User}}/.gitconfig' {{.User}}
+  - {{endTaskCheck .}}
+
+{{- if not (.MountPath "devstack")}}
+  - {{beginTask . "Download devstack" }}
+  - sudo -u {{.User}} {{template "ENV" .}} git clone http://git.openstack.org/openstack-dev/devstack {{template "DEVSTACKPATH" .}}
+  - {{endTaskCheck .}}
+
+# Once dyamic IP provisioning is available, the HOST_IP can be populated with
+# it against the current static one.
+  - {{beginTask . "Create conf file" }}
+  - su -c 'printf "%s\n"
+    "[[local|localrc]]"
+    "HOST_IP=127.0.0.1"
+    "ADMIN_PASSWORD=secret"
+    "DATABASE_PASSWORD=\$ADMIN_PASSWORD"
+    "RABBIT_PASSWORD=\$ADMIN_PASSWORD"
+    "SERVICE_PASSWORD=\$ADMIN_PASSWORD"
+    > {{template "DEVSTACKPATH" .}}/local.conf' {{.User}}
+  - {{endTaskCheck .}}
+{{ end }}
+
+  - {{beginTask . "Stacking the system" }}
+  - su -c "cd {{template "DEVSTACKPATH" .}} && nohup ./stack.sh > /home/{{.User}}/stack.log &" {{.User}}
+  - {{endTaskCheck . }}
+
+# Will enable this once the other PR is merged. Very crude this way, but github does not support PR dependencies.
+# {{/*messageHost . "Follow stack.log in $HOME to see output of stacking process" */}}
```

Comment:
We could provide the user with the exact command to run, e.g.,

{{message . (printf "ccloudvm run %s \"tail -f /home/%s/stack.log\"" .Hostname .User)}}

Note I haven't checked this actually works, particularly the quoting.